### PR TITLE
Prepare some Geometry ESProducers for concurrent IOVs

### PR DESCRIPTION
**Pull request by wddgit**
The following types of changes are included:
    - changing return type of produce to unique_ptr
    - removing unnecessary "dependsOn" callbacks
    - some callback content moved to produce
    - 4 callbacks were simply empty (did nothing)
    - some objects do not need to be cached create on stack
    - removing unneeded data members, includes, declarations
    - fix potential assert failure if produce called more than once
    - fix potential double delete if produce called more than once
    - 2 memory leaks
    - some general cleanup deleting unnecessary things

### CODE REVIEW

**Comment by wddgit** [comment]
please test

File: Geometry/HcalCommonData/plugins/HcalDDDSimConstantsESModule.cc [hunk 1]
```diff
@@ -23,36 +23,25 @@
...
-//#define EDM_ML_DEBUG
-
 class HcalDDDSimConstantsESModule : public edm::ESProducer {
 
 public:
   HcalDDDSimConstantsESModule(const edm::ParameterSet&);
   ~HcalDDDSimConstantsESModule() override;
 
-  typedef std::shared_ptr<HcalDDDSimConstants> ReturnType;
+  typedef std::unique_ptr<HcalDDDSimConstants> ReturnType;
```
Comment (ianna) [inline Geometry/HcalCommonData/plugins/HcalDDDSimConstantsESModule.cc]:
```
using ReturnType = std::unique_ptr<HcalDDDSimConstants>;
```

File: Geometry/EcalMapping/plugins/EcalElectronicsMappingBuilder.h [hunk 2]
```diff
@@ -9,22 +9,18 @@
 #include "FWCore/Framework/interface/ModuleFactory.h"
 #include "FWCore/Framework/interface/ESProducer.h"
 
-
-// class EcalMappingRcd;
-
-
 // #include "FWCore/Framework/interface/EventSetupRecordIntervalFinder.h"
```
Comment (ianna) [inline Geometry/EcalMapping/plugins/EcalElectronicsMappingBuilder.h]: please, remove commented out includes

File: Geometry/EcalMapping/plugins/EcalElectronicsMappingBuilder.h [hunk 3]
```diff
@@ -9,22 +9,18 @@
...
-
-#include "FWCore/Framework/interface/ESHandle.h"
-#include "FWCore/ParameterSet/interface/ParameterSet.h"
 // #include "Geometry/Records/interface/IdealGeometryRecord.h"
```
Comment (ianna) [inline Geometry/EcalMapping/plugins/EcalElectronicsMappingBuilder.h]: please, remove commented out includes

File: Geometry/EcalMapping/plugins/EcalElectronicsMappingBuilder.h [hunk 4]
```diff
@@ -41,14 +37,10 @@ class EcalElectronicsMappingBuilder : public edm::ESProducer
...
   
-  void DBCallback (const EcalMappingElectronicsRcd& fRecord);
-  
  private:
   void FillFromDatabase(const std::vector<EcalMappingElement>& ee,
                         EcalElectronicsMapping& theMap);
   
-  
-  const EcalMappingElectronics* Mapping_ ;
   // void setIntervalFor(const edm::eventsetup::EventSetupRecordKey &, const edm::IOVSyncValue&, edm::ValidityInterval & );
```
Comment (ianna) [inline Geometry/EcalMapping/plugins/EcalElectronicsMappingBuilder.h]: please, remove commented out code

File: Geometry/HcalEventSetup/interface/HcalDDDGeometryEP.h [hunk 5]
```diff
@@ -27,18 +27,15 @@ class HcalDDDGeometryEP : public edm::ESProducer {
   HcalDDDGeometryEP(const edm::ParameterSet&);
   ~HcalDDDGeometryEP() override;
 
-  typedef std::shared_ptr<CaloSubdetectorGeometry> ReturnType;
+  typedef std::unique_ptr<CaloSubdetectorGeometry> ReturnType;
```
Comment (ianna) [inline Geometry/HcalEventSetup/interface/HcalDDDGeometryEP.h]:
```
using ReturnType = std::unique_ptr<CaloSubdetectorGeometry>;
```

File: Geometry/HcalEventSetup/interface/HcalHardcodeGeometryEP.h [hunk 6]
```diff
@@ -19,13 +19,11 @@ class HcalHardcodeGeometryEP : public edm::ESProducer {
   HcalHardcodeGeometryEP(const edm::ParameterSet&);
   ~HcalHardcodeGeometryEP() override;
 
-  typedef std::shared_ptr<CaloSubdetectorGeometry> ReturnType;
+  typedef std::unique_ptr<CaloSubdetectorGeometry> ReturnType;
```
Comment (ianna) [inline Geometry/HcalEventSetup/interface/HcalHardcodeGeometryEP.h]:
```
using ReturnType = std::unique_ptr<CaloSubdetectorGeometry>;
```

File: Geometry/HcalEventSetup/interface/HcalHardcodeGeometryEP.h [hunk 7]
```diff
@@ -19,13 +19,11 @@ class HcalHardcodeGeometryEP : public edm::ESProducer {
...
 
   ReturnType produceIdeal(const HcalRecNumberingRecord&);
   ReturnType produceAligned(const HcalGeometryRecord& );
 
-  void       idealRecordCallBack(const HcalRecNumberingRecord&) {}
-
 private:
   edm::ParameterSet ps0;
```
Comment (ianna) [inline Geometry/HcalEventSetup/interface/HcalHardcodeGeometryEP.h]:
```
edm::ParameterSet ps0_;
```

File: Geometry/HcalEventSetup/interface/HcalTopologyIdealEP.h [hunk 8]
```diff
@@ -28,18 +28,15 @@ class HcalTopologyIdealEP : public edm::ESProducer {
   HcalTopologyIdealEP(const edm::ParameterSet&);
   ~HcalTopologyIdealEP() override;
 
-  typedef std::shared_ptr<HcalTopology> ReturnType;
+  typedef std::unique_ptr<HcalTopology> ReturnType;
```
Comment (ianna) [inline Geometry/HcalEventSetup/interface/HcalTopologyIdealEP.h]:
```
using ReturnType = std::unique_ptr<HcalTopology>;
```

File: Geometry/MuonNumbering/plugins/MuonNumberingInitialization.cc [hunk 9]
```diff
@@ -39,19 +35,11 @@ class MuonNumberingInitialization : public edm::ESProducer
   typedef std::unique_ptr<MuonDDDConstants> ReturnType;
```
Comment (ianna) [inline Geometry/MuonNumbering/plugins/MuonNumberingInitialization.cc]:
```
using ReturnType = std::unique_ptr<MuonDDDConstants>;
```

**Review by ianna — CHANGES_REQUESTED** [review]

**Comment by wddgit** [comment]
please test
I pushed a new fix that cleans up all the problems Ianna suggested I fix. Thanks for the suggestions. When I got going I also fixed some other similar things. So the second commit includes:

      - deleting commented code and useless comments
      - deleting trailing whitespace
      - converting from typedef to using statements
      - removing an unused copy of a ParameterSet
      - removing unneeded blank lines

Unless I made a typographical blunder, none of this should effect behavior in any way, although it does make the code easier to read.

**Comment by ianna** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**